### PR TITLE
SONAR-8857 allow same name for two Quality Profiles in different organisations

### PR DESCRIPTION
**Pull request by bartfastiel**

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java [hunk 1]
```diff
@@ -172,6 +172,34 @@ public void create_profile_for_specific_organization() {
...
+    // this name will be used twice
+    String profileName = "Profile123";
+
+    OrganizationDto organization1 = dbTester.organizations().insert();
+    TestRequest request1 = wsTester.newRequest()
+      .setMediaType(MediaTypes.PROTOBUF)
+      .setParam("organization", organization1.getKey())
+      .setParam("name", profileName)
+      .setParam("language", XOO_LANGUAGE)
+      .setParam("backup_with_messages", "<xml/>");
```
Comment (simonbrandhof) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: You should try to drop this param

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java [hunk 2]
```diff
@@ -169,7 +169,7 @@ public QualityProfileDto selectByNameAndLanguage(OrganizationDto organization, S
    */
   @Deprecated
   public List<QualityProfileDto> selectByNameAndLanguages(String name, Collection<String> languageKeys, DbSession session) {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java [hunk 3]
```diff
@@ -156,7 +156,7 @@ public QualityProfileDto selectParentById(DbSession session, int childId) {
   @Deprecated
   @CheckForNull
   public QualityProfileDto selectByNameAndLanguage(String name, String language, DbSession session) {
```
Comment (SonarTech) [inline server/sonar-db-dao/src/main/java/org/sonar/db/qualityprofile/QualityProfileDao.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: Done. Thanks.

**Pull request merged**